esp/clients/roxiecontrol.cpp

```diff
@@ -111,3 +153,37 @@ IPropertyTree *sendRoxieControlAllNodes(const SocketEndpoint &ep, const char *ms
     Owned<ISocket> sock = ISocket::connect_timeout(ep, wait);
     return sendRoxieControlAllNodes(sock, msg, allOrNothing, wait);
 }
+
+IPropertyTree *sendSecureRoxieControlAllNodes(ISmartSocketFactory *conn, const char *msg, bool allOrNothing, unsigned wait, unsigned connect_wait, bool publicService)
+{
+    const SocketEndpoint &ep = conn->nextEndpoint();
```

Comment:
most f this code looks identical to the code in sendRoxieControlQuery(), and a candidate for being shared.

Reply:
if conn->isTlsService() was checked within this function it would probably simplify the calling logic.